tests/subsys/tracing/src/main.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ztest.h>
+
+/**
+ * @defgroup subsys_tracing_tests subsys tracing
+ * @ingroup all_tests
+ * @{
+ * @}
+ */
+
+/* size of stack area used by each thread */
+#define STACK_SIZE (512 + CONFIG_TEST_EXTRA_STACKSIZE)
+#define LOOP_TIMES 200
+
+#define dummy_test(_name) \
+	static void _name(void) \
+	{ \
+		ztest_test_skip(); \
+	}
+
+static struct k_thread thread;
+K_THREAD_STACK_DEFINE(thread_stack, STACK_SIZE);
+
+/* define semaphores */
+K_SEM_DEFINE(thread1_sem, 1, 1);
+K_SEM_DEFINE(thread2_sem, 0, 1);
+
+bool trace_thread_invoked;
+bool trace_ISR_invoked;
+bool trace_thread_idle_invoked;
+
+/* thread handle for switch */
+void thread_handle(void *p1, void *self_sem, void *other_sem)
+{
+	for (int i = 0; i < LOOP_TIMES; i++) {
+		/* take my semaphore */
+		k_sem_take(self_sem, K_FOREVER);
+
+		/* wait for a while, then let other thread have a turn. */
+		k_sleep(K_MSEC(100));
+
+		k_sem_give(other_sem);
+	}
+}
+
+/* spawn a thread to generate more tracing data about thread switch. */
+void create_handle_thread(void)
+{
+	k_thread_create(&thread, thread_stack, STACK_SIZE,
+		thread_handle, NULL, &thread2_sem, &thread1_sem,
+		K_PRIO_PREEMPT(0), K_INHERIT_PERMS,
+		K_NO_WAIT);
+
+	thread_handle(NULL, &thread1_sem, &thread2_sem);
+}
+
+/**
+ * we need to use external tools to handle tracing data
+ * which produced by qemu/posix platform with different
+ * backends. and use popen api to execute the tools, so
+ * below test cases only avalible on posix arch.
+ */
+#ifdef CONFIG_ARCH_POSIX
+#include <unistd.h>
+#include <stdlib.h>
+#define MAX_BYTES 1024
+char result_buf[MAX_BYTES];
+
+/**
+ * @brief Test if tracing support thread, ISR, idle thread operation.
+ *
+ * @details
+ * - The consumer of the tracing data shall be able to identify and show
```

Comment:
sound like a requirements, does not give any details

Reply:
I will add detail info and change the code, thanks for your comments.